platform/ios/app/MBXViewController.m

```diff
@@ -227,37 +230,34 @@ @implementation MBXViewController
 
 #pragma mark - Setup & Teardown
 
-+ (void)initialize
-{
-    if (self == [MBXViewController class])
-    {
-        [[NSUserDefaults standardUserDefaults] registerDefaults:@{
-            @"MBXUserTrackingMode": @(MGLUserTrackingModeNone),
-            @"MBXShowsUserLocation": @NO,
-            @"MBXDebug": @NO,
-        }];
-    }
-}
-
 - (void)viewDidLoad
 {
     [super viewDidLoad];
 
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(saveState:) name:UIApplicationDidEnterBackgroundNotification object:nil];
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(restoreState:) name:UIApplicationWillEnterForegroundNotification object:nil];
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(saveState:) name:UIApplicationWillTerminateNotification object:nil];
+    // Keep track of current map state and debug preferences,
+    // saving and restoring when the application's state changes.
+    self.currentState =  [MBXStateManager sharedManager].currentState;
+
+    if (!self.currentState) {
```

Comment:
You should restore the previous state if `self.currentState` is not `nil` since you are not using directly the properties in `currentState`:
e.g.
```
if(!self.currentState) {
...
} else {
self.mapView. showsUserHeadingIndicator =  self.currentState. showsUserHeadingIndicator;
...
}
```